### PR TITLE
Ff clean code

### PR DESCRIPTION
**Pull request by jonp-h**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactor and clean code in fish factory

* **What is the current behavior?** (You can also link to an open issue here)
Some code is unclear and does not follow a clear convention
#387 

* **What is the new behavior?** (if this is a feature change)
Add XML comments and clarify most code. Refactor and simplify some functions. Renamed unclear filenames and functions in gillcutting and badcutting.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
DiscardBadFish.cs, TaskPointListener.cs and BarrierScript.cs remains untouched

### CODE REVIEW

File: Assets/FishFactory/Scripts/stunFish.cs
Comment (mefriis) [inline Assets/FishFactory/Scripts/stunFish.cs:1]: Filename should start with capital S?

**Review by Herman1412 — APPROVED** [review]
LGTM

File: Assets/FishFactory/Scripts/FactoryFishState.cs
Comment (krisgjol) [inline Assets/FishFactory/Scripts/FactoryFishState.cs:1]: line 33 plays wrong sound?

File: Assets/FishFactory/Scripts/FactoryFishSpawner.cs
Comment (krisgjol) [inline Assets/FishFactory/Scripts/FactoryFishSpawner.cs:1]: problm with the percentages is that there is no check to make sure it all adds up to 100? you can set them both 100% leaving 0 chance for a stunned fish, and probably messing with the chances of the other two working as well?

**Review by Siggiboyy — APPROVED** [review]
Fine kommentarer, mye mer oversiktelig enn før

**Review by krisgjol — APPROVED** [review]

File: Assets/FishFactory/Scripts/FactoryFishState.cs
Comment (jonp-h) [reply Assets/FishFactory/Scripts/FactoryFishState.cs:1]: it is correct

**Pull request merged**